Box_intersection_d/include/CGAL/Box_intersection_d/Box_d.h

```diff
@@ -36,14 +38,16 @@ namespace Box_intersection_d {
 template <class Dummy>
 struct Unique_numbers {
     typedef std::size_t ID;
-    Unique_numbers() : i(n++) {}
+    Unique_numbers() {
+      static CGAL::cpp11::atomic<std::size_t> n; // initialized to 0
+      i = n++;
+    }
```

Comment:
As `n` is now an `atomic<integral>`, [its post-increment `operator++ is equivalent to `n.fetch_add(1)`][*]. That synchronizes with other threads.

[*]: http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith